index.js

```diff
@@ -37,7 +37,9 @@ module.exports = function (h, opts) {
           } else {
             p.push([ OPEN, arg ])
           }
-        } else {
+        } else if (xstate === COMMENT && opts.comments) {
+          reg += String(arg)
+        } else if (xstate !== COMMENT) {
```

Comment:
What happens if `xstate === COMMENT` and `opts.comments` is falsey?

Is it okay for that condition to be unhandled?

Reply:
yes—in that case, the comment contents were discarded, and we have to discard the embedded value as well. previously, without this check, the embedded value would be added as a text node and visible in the output.